### PR TITLE
Galaxy Introduction hands-on manual

### PR DESCRIPTION
**Pull request by shiltemann**
still a work in progress, heading home now, will continue tomorrow, but if anybody has any suggestions or comments let me know!

### CODE REVIEW

**Comment by bgruening** [comment]
I like it very much. One thing we need to discuss in general is how to name these tutorials. Your name is now pretty generic `Introduction.md` and they clash with the ones already here https://github.com/bgruening/training-material/tree/master/Introduction/tmp.

Any idea how we can handle this?

**Comment by shiltemann** [comment]
easy solution would be for me to just rename to something less generic like `Galaxy101.md`, or if we want names unique across topics as well we could prefix with topic name, so `Introduction_Galaxy101.md` or something similar?

**Comment by bgruening** [comment]
I admit the `tmp` is confusing, we will change this. The initial idea was to put things there and work on them to move them to the proper place later. But people seem to ignore it, what is understandable. The name Galaxy101 also exists.

I guess we need to add a small section in the main introduction-readme about all the tutorials and there content and requirements.

ping @bebatut

**Comment by shiltemann** [comment]
ah gotcha, I just started working on the file named Introduction.md in main directory because it already had the skeleton for the proper style ;P ..and assumed tmp dir was just gathered existing materials for reference, but this makes sense, well, intro should be done tomorrow, next time will work in tmp directory :) ..and needless to say I am happy to change filename to what seems appropriate

yep, having a small description for instructors of what it takes to run this tutorial would be nice, for those who don't want to use the docker but install everything for a tutorial on an existing galaxy for instance, overview of what tools and reference data they need to install etc, what features to enable, minimal version of Galaxy, things like that..

**Comment by shiltemann** [comment]
P.S. yep, this tutorial is basically an adaptation of the old Galaxy101 tutorial, just with more/updated screenshots and some extra text where we've noticed people often had questions :)

File: Introduction/tutorials/Introduction.md [hunk 1]
```diff
@@ -1,97 +1,269 @@
...
+3. Click **Execute** and examine the output file, this should contain only the first 5 lines of the previous dataset.
+
+## Recovering exon info and displaying data in genome browsers
+
+Congratulations! you have now determined which exons on chromosome 22 have the highest number of SNPS, but what else can we learn about them? One way to learn more about a genetic location is to view it in a genome browser. However, in the process of getting our answer, we have lost information about the location of these exons on the chromosome. But fear not, Galaxy never deletes any of your data, so we can recover this information back into our file quite easily.
+
+:pencil2: ***Hands on!***
+
+1. Open the tool `Compare two Datasets - to find common or distinct rows`
+2. Set the parameters to compare the column 4 of the exon file with column one of the top-5 exons file to find mathing rows.
```
Comment (nsoranzo) [inline Introduction/tutorials/Introduction.md]: s/mathing/matching/

File: Introduction/tutorials/Introduction.md [hunk 2]
```diff
@@ -1,97 +1,269 @@
...
+
+6. This will upload the data to UCSC as custom track. To see your data look at the `User Supplied Track` near the top. You can enter the coordinates of one of your exons at the top to jump to that location.
+
+ ![](../images/101_21.png)
+
+UCSC provides a large number of tracks that can help you get a sense of your genomic area, it contains common SNPS, repeats, genes, and much much more (scroll down to find all possible tracks)
+
+## Understanding histories
+
+In Galaxy your analyses live in histories such as your current one. Histories can be very large, you can have as many histories as you want. You can control your histories (switching, copying, sharing, creating a fresh history, etc) in the **Options** menu on the top of the history pane (gear symbol):
```
Comment (nsoranzo) [inline Introduction/tutorials/Introduction.md]: > Histories can be very large, you can have as many histories as you want.

I'd add a `but` after the comma.

**Comment by nsoranzo** [comment]
@shiltemann Fantastic work, thanks!

**Comment by shiltemann** [comment]
thanks for the review @nsoranzo! :)

**Comment by shiltemann** [comment]
ok, added the rest of the tutorial now, please review :)

**Comment by bgruening** [comment]
Yeah, yeah!!!

File: Introduction/tutorials/Introduction.md [hunk 3]
```diff
@@ -1,106 +1,404 @@
...
+
+## Preparations
+
+Time to get started, browse to your Galaxy instance and log in or register. The Galaxy interface consist of three main parts. The available tools are listed on the left, your analysis history is recorded on the right, and the middle pane will show the tools and datasets.
+
+![](../images/galaxy-interface.png)
+
+:pencil2: ***Hands on!***
+
+1. Make sure you have an empty analysis history. To start a new history, click on the **gear icon** at the top of the history pane, and select the option **Create New** from the menu.
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: With @blankclemens, we propose to format the tip, like this:
```
    > :bulb: **Starting a new history**
    > * **gear icon** at the top of the history panel
    > * select the option **Create New** from the menu
    <br>
```
What do you think?

BTW, I think there is an "l" missing in "panel"

File: Introduction/tutorials/Introduction.md [hunk 4]
```diff
@@ -1,106 +1,404 @@
...
+2. **Rename your history** to make it easy to recognize. You can do this by clicking on the title of the history (by default the title is *Unnamed history*) and typing **Galaxy 101** as the name.
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: You named it as the previous tutorial?

**Comment by shiltemann** [comment]
any final thoughts on the file name @bgruening? Introduction? Galaxy101? MyFirstAnalysis? something else?

File: Introduction/tutorials/Introduction.md [hunk 5]
```diff
@@ -1,106 +1,404 @@
...
+
+Now we are ready to do some analysis, but first we will need to get some data into our history. You can upload files from your computer, but Galaxy can also fetch data directly from external sources. We will now import a list of all the exon locations on chromosome 22 directly from the UCSC table browser.
+
+:pencil2: ***Hands on!***
+
+1. In the tool menu, navigate to `Get Data -> UCSC Main - table browser`
+
+ ![](../images/101_01.png)
+
+2. You will be taken to the **UCSC table browser**, which looks something like this:
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: It is not really a step there? Maybe you can just change the `2. ` to a space

File: Introduction/tutorials/Introduction.md [hunk 6]
```diff
@@ -1,106 +1,404 @@
...
+
+Now we have information about the exon locations, but our question was which exon contains the largest number of SNPs, so let's get some information about SNP locations from UCSC as well:
 
 :pencil2: ***Hands on!***
 
-1. First step
-2. Second step
-3. Third step
+1. Return to the UCSC tool `Get Data -> UCSC Main - table browser`
+2. This time change the setting in **group** to `Variation`
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: The "This time" may be removed to have only "affirmative" sentence

File: Introduction/tutorials/Introduction.md [hunk 7]
```diff
@@ -1,106 +1,404 @@
...
+
+ ![](../images/101_06.png)
+
+3. The **track** setting shows the version of the SNP database to get. In this example it is version 142, but you may select the latest one. Your results may vary slightly from the ones in this tutorial when you select a different version, but in general it is a good idea to select the latest version, as this will contain the most up-to-date SNP information.
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: Maybe it is not really a step here

File: Introduction/tutorials/Introduction.md [hunk 8]
```diff
@@ -1,106 +1,404 @@
...
+4. To **view the contents** of the file, click on the **eye icon**. Your file should look something like this:
 
-# Part 1
+ ![](../images/101_exons.png)
 
-Introduction about this part
+ Each line represents an exon, the first three columns are the genomic location, and the fourth column contains the ID (name) of the exon.
 
-## Subpart 1
+5. Let's rename our dataset to something more recognizable. Click on the **pencil icon** to edit a file's attributes, in the next screen change the name of the dataset to `Exons`, then click the **Save** button at the bottom of the screen. Your history should now look something like this:
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: Maybe sublist here. What do you think?

File: Introduction/tutorials/Introduction.md [hunk 9]
```diff
@@ -1,106 +1,404 @@
...
+# Finding Exons with the highest number of SNPs
+
+Let’s remind ourselves that our objective was to find which exon contains the most SNPs. This  first step in answering this question will be joining the file with the exon locations with the file containing the SNP locations (here `join` is just a fancy word for printing the SNPs and exons that overlap side-by-side).
 
 :pencil2: ***Hands on!***
 
-1. First step
-2. Second step
-3. Third step
+1. Different Galaxy servers may have tools available under different sections, therefore it is often useful to use the **search bar** at the top of the tool panel to find your tool. Enter the word `join` in the search bar of the tool panel, and select the tool named `Join - the intervals of two datasets side-by-side`
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: Maybe moving the first sentence above the hands on?

File: Introduction/tutorials/Introduction.md [hunk 10]
```diff
@@ -1,106 +1,404 @@
...
+
+3. Click the **Execute** button and view the resulting file (with the eye icon). If everything went okay, you should see a file that looks similar to this:
+
+ ![](../images/101_joined.png)
+
+ Remember that variations are possible due to using different versions of UCSC databases, as long as you have similar looking columns you did everything right :)
+
+Let’s take a look at this dataset. The first six columns correspond to the exons, and the last six columns correspond to the SNPs. Column 4 contains the exon IDs, and column 10 contains the SNP IDs. In our screenshot you see that the first 5 lines in the file all have the same exon ID (`uc010gqp.2_cds_10_0_chr22_16287254_r`) but different SNP IDs, this means these lines represent 5 different SNPs that all overlapped with the same exon. So we can find the total number of SNPs in an exon simply by counting the number of lines that have the exon ID in the fourth column.
+
+:question: Question: for the first 3 exons in your file, what is the number of SNPs that fall into that exon?
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: If there is the question emoji, there is maybe no need for "Question:" word. What do you think?

**Comment by bgruening** [comment]
`introduction_my_first_analysis.md`? :)

File: Introduction/tutorials/Introduction.md [hunk 11]
```diff
@@ -1,106 +1,404 @@
...
+
+2. Set the parameters to compare the column 4 of the exon file with column 1 of the top-5 exons file to find matching rows.
+
+ ![](../images/101_18.png)
+
+3. Click **Execute** and examine your output file. It should contain the locations of your top 5 exons:
+
+ ![](../images/101_19.png)
+
+4. A good way to learn about these exons is to look at their genomic surrounding. This can be done by using genome browsers. First, we have to tell Galaxy which **Genome build** this data uses (hg19), we can do this as follows:
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: Maybe the first sentence can be moved out of this list and a new hands on started. What do you think?

File: Introduction/tutorials/Introduction.md [hunk 12]
```diff
@@ -1,106 +1,404 @@
...
+
+ ![](../images/101_20.png)
+
+5. To **visualize the data in UCSC genome browser**, click on `display at UCSC main` option visible when you expand the history item.
+
+ ![](../images/101_displayucsc.png)
+
+6. This will upload the data to UCSC as custom track. To see your data look at the `User Track` near the top. You can enter the coordinates of one of your exons at the top to jump to that location.
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: Maybe in the previous number as comment?

**Comment by shiltemann** [comment]
done :)

File: Introduction/tutorials/Introduction.md [hunk 13]
```diff
@@ -1,106 +1,404 @@
...
+
+UCSC provides a large number of tracks that can help you get a sense of your genomic area, it contains common SNPS, repeats, genes, and much much more (scroll down to find all possible tracks)
+
+## Understanding histories
+
+In Galaxy your analyses live in histories such as your current one. Histories can be very large, and you can have as many histories as you want. You can control your histories (switching, copying, sharing, creating a fresh history, etc) in the **Options** menu on the top of the history pane (gear symbol):
+
+![](../images/101_22.png)
+
+If you create a new history, your current history does not disappear. If you would like to list all of your histories just choose `Saved Histories` from the history menu and you will see a list of all your histories in the center pane:
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: A "l" is missing in "panel"

File: Introduction/tutorials/Introduction.md [hunk 14]
```diff
@@ -1,106 +1,404 @@
...
+
+ ![](../images/101_29.png)
+
+ If you click on this asterisk for any of the output datasets, then *only* files with an asterisk will be shown, and all outputs without an asterisk will be hidden. (Note that clicking *all* outputs has the same effect as clicking *none* of the outputs, in both cases all the datasets will be shown.)
+
+3. **Click the asterisk** next to `out_file1` in the `Select First` and `Compare two Datasets` tools. Now when we run the workflow, we will only see the final two outputs, our list with the top-5 exons and their SNP counts, and the file with exons ready for viewing in a genome browser. Once you have done this, you will notice that the **minimap** at the bottom-right corner of your screen will have a colour-coded view of your workflow, with orange boxes representing a tool with an output that will be shown.
```
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: Maybe moving after "`Compare two Datasets` tools" as a comment

File: Introduction/tutorials/Introduction.md (hunk 3 above)
Comment (shiltemann) [inline Introduction/tutorials/Introduction.md]: pane is a word too, means basically the same as panel (but if peopole think panel is more widely understood I can change it)

**Comment by shiltemann** [comment]
thanks for reviewing @bebatut! :)

File: Introduction/tutorials/Introduction.md (hunk 8 above)
Comment (shiltemann) [inline Introduction/tutorials/Introduction.md]: good idea!

File: Introduction/tutorials/Introduction.md (hunk 3 above)
Comment (bebatut) [inline Introduction/tutorials/Introduction.md]: I did not know. I will sleep less stupid tonight ;) Thanks for the info

File: Introduction/tutorials/Introduction.md (hunk 10 above)
Comment (shiltemann) [inline Introduction/tutorials/Introduction.md]: to me it looked a bit weird without, as if it maybe would not be clear they needed to answer this? but I do not care really strongly either way :)

Comment (shiltemann) [inline Introduction/tutorials/Introduction.md]: ok, in github looks a lot better than my local preview :P will change, thanks :)

File: Introduction/tutorials/Introduction.md (hunk 3 above)
Comment (shiltemann) [inline Introduction/tutorials/Introduction.md]: love the tip idea, thanks :)

**Comment by bgruening** [comment]
Great work!!!

**Pull request merged**